index.bs

```diff
@@ -3199,6 +3511,176 @@ The [=remote end steps=] with |command parameters| are:
 1. Return a new map matching the <code>ScriptEvaluateResultSuccess</code>
    production, with the <code>result</code> field set to |result|.
 
+#### The script.callFunction Command ####  {#command-script-callFunction}
+
+The <dfn export for=commands>script.callFunction</dfn> command calls a provided
+function with given arguments in a given realm.
+
+<code>RealmInfo</code> can be either a realm or a browsing context.
+
+Note: In case of an arrow function in <code>functionDeclaration</code>, the <code>this</code>
+argument doesn't affect function's <code>this</code> binding.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      ScriptCallFunctionCommand = {
+        method: "script.callFunction",
+        params: ScriptCallFunctionParameters
+      }
+
+      ScriptCallFunctionParameters = {
+        functionDeclaration: text;
+        arguments: [ArgumentValue];
+        ?this: ArgumentValue;
+        ?awaitPromise: bool;
+        target: Target;
+      }
+
+      ArgumentValue = (
+        RemoteReference //
+        LocalValue
+      );
+
+    </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+      ScriptCallFunctionResult = (
+        ScriptCallFunctionResultSuccess //
+        ScriptCallFunctionResultException
+      )
+
+      ScriptCallFunctionResultSuccess = {
+         result: RemoteValue
+      }
+
+      ScriptCallFunctionResultException = {
+        exceptionDetails: ExceptionDetails
+      }
+    </pre>
+   </dd>
+</dl>
+
+Issue: TODO: Add timeout argument as described in the script.evaluate.
+
+<div algorithm>
+To <dfn>deserialize a parameter value</dfn> given an |serialized value|:
+
+1. If |serialized value| matches the [=RemoteReference=] production, return
+   [=deserialize remote reference=] of |serialized value|.
```

Comment:
Ah, so here we actually check for the remote reference. Should we repeat that for each and every command, or better handle it within `[=deserialize local value=]`?

Reply:
`[=deserialize local value=]` now cares about `[=RemoteReference=]`